bundles/org.openhab.binding.miio/src/main/java/org/openhab/binding/miio/internal/cloud/MiCloudConnector.java

```diff
@@ -0,0 +1,433 @@
+        return "";
+    }
+
+    public String getDeviceStatus(String device, String country) throws MiCloudException {
+        String url = getApiUrl(country) + "/home/device_list";
+        Map<String, String> map = new HashMap<String, String>();
+        map.put("data", "{\"dids\":[\"" + device + "\"]}");
+        final String response = request(url, map);
+        logger.debug("response: {}", response);
+        return response;
+    }
+
+    public String getDevices(String country) {
+        String url = getApiUrl(country) + "/home/device_list";
+        Map<String, String> map = new HashMap<String, String>();
+        map.put("data", "{\"getVirtualModel\":false,\"getHuamiDevices\":0}");
+        String resp;
+        try {
+            resp = request(url, map);
+            logger.trace("Get devices response: {}", resp);
+            CloudUtil.printDevices(resp, country, logger);
+            if (resp.length() > 2) {
+                CloudUtil.saveFile(resp, country, logger);
+                return resp;
+            }
+        } catch (MiCloudException e) {
+            logger.info("{}", e.getMessage());
+        }
+        return "";
+    }
+
+    public String request(String urlPart, String country, Map<String, String> params) throws MiCloudException {
+        String url = getApiUrl(country) + urlPart;
+        String response = request(url, params);
+        logger.debug("Request to {} server {}. Response: {}", country, urlPart, response);
+        return response;
+    }
+
+    public String request(String url, Map<String, String> params) throws MiCloudException {
+        if (this.serviceToken.isEmpty() || this.userId.isEmpty()) {
+            throw new MiCloudException("Cannot execute request. service token or userId missing");
+        }
+        startClient();
+        logger.debug("Send request: {} to {}", params.get("data"), url);
+        Request request = httpClient.newRequest(url).timeout(REQUEST_TIMEOUT_SECONDS, TimeUnit.SECONDS);
+        request.agent(USERAGENT);
+        request.header("x-xiaomi-protocal-flag-cli", "PROTOCAL-HTTP2");
+        request.header(HttpHeader.CONTENT_TYPE, "application/x-www-form-urlencoded");
+        request.cookie(new HttpCookie("userId", this.userId));
+        request.cookie(new HttpCookie("yetAnotherServiceToken", this.serviceToken));
+        request.cookie(new HttpCookie("serviceToken", this.serviceToken));
+        request.cookie(new HttpCookie("locale", locale.toString()));
+        request.cookie(new HttpCookie("timezone", "GMT%2B01%3A00"));
+        request.cookie(new HttpCookie("is_daylight", "1"));
+        request.cookie(new HttpCookie("dst_offset", "3600000"));
+        request.cookie(new HttpCookie("channel", "MI_APP_STORE"));
+
+        for (HttpCookie cookie : request.getCookies()) {
+            logger.trace("Cookie set for request ({}) : {} --> {}     (path: {})", cookie.getDomain(), cookie.getName(),
+                    cookie.getValue(), cookie.getPath());
+        }
```

Comment:
Just minimizing cpu waste.
```suggestion
        if(logger.isTraceEnabled){
        for (HttpCookie cookie : request.getCookies()) {
            logger.trace("Cookie set for request ({}) : {} --> {}     (path: {})", cookie.getDomain(), cookie.getName(),
                    cookie.getValue(), cookie.getPath());
        }
        }
```